ui/components/data-sources/BaseDataSourceDialog.js

```diff
@@ -67,22 +67,17 @@ class BaseDataSourceDialog extends Component {
 
         // Set the initial validation state and assume that port is valid
         // (because we set a default value there)
-        const validationDetails = {
-            url: "error",
-            database: "error",
-            username: "error",
-            port: "success"
-        };
-
-        Validate([
+        const postgresDetails = {};
+
+        const options = Validate([
             Validators.URL.valid, url, "url",
             Validators.String.nonBlank, database, "database",
             Validators.String.nonBlank, username, "username",
-            Validators.Port.valid, port, "port"
-        ], validationDetails);
+            Validators.Number.natural, port, "port"
+        ], postgresDetails);
 
         const { validations } = this.state;
-        const newValidations = { ...validations, ...validationDetails };
+        const newValidations = { ...validations, postgresDetails, options };
```

Comment:
What is `options` and why don't we include those under `postgresDetails`?

Reply:
`options` is the overall validation result that determines if the `Ok` button will be disabled. `postgresOptions` contains the validation results for the postgres options and is only used to display the validation state in the postgres options component. We can't include them in `validations` directly because based on the selected type they are available or not.